tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py

```diff
@@ -324,6 +324,32 @@ def _test_same_keeper(self) -> None:
         assert self.hardhat_provider.make_request(
             RPCEndpoint("evm_setIntervalMining"), [mining_interval_msecs]
         ), "Re-enabling auto-mining failed!"
+        """
+        NOTE:
+        The second gas estimation will ALWAYS fail. That is, because the
+        current implementation of the plugin does not play well with Hardhat:
+        https://github.com/valory-xyz/open-aea/blob/main/plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py#L1113-L1118
+        https://github.com/NomicFoundation/hardhat/blob/d1733f9ecfe8125111a707ff0e3dea287e584caa/packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts#L419-L424
+        eth_estimateGas
+          Contract call:       GnosisSafeProxy#<unrecognized-selector>
+          From:                0xbcd4042de499d14e55001ccbb24a551f3b954096
+          To:                  0x68fcdf52066cce5612827e872c45767e5a1f6551
+          Value:               0 ETH
+
+          Error: VM Exception while processing transaction: reverted with reason string 'GS026'
```

Comment:
https://github.com/safe-global/safe-contracts/blob/main/docs/error_codes.md this indicates that the signature is incorrect, which here is due to the nonce (which would be invalid if the first transaction ever got mined first)

Reply:
In my understanding the error code is correct, the nonce is increased and the signature is invalid in the pending state (block). The problem is that hardhat doesn't support changing with which state of the blockchain the estimation should be done, i.e. setting the block identifier. This is also true for some other eth node implementations.

Reply:
We are saying mostly the same in different ways. From the perspective of tx execution the second tx is invalid of the first tx has been committed (which is what happens when gas estimation is run in pending state). The invalidity stems from the nonce on the safe side (not the tx nonce!) being identical for both txs.

Reply:
I was referring to the safe nonce.